elasticapm/traces.py

```diff
@@ -64,17 +118,42 @@ def __init__(self, tracer, transaction_type="custom", trace_parent=None, is_samp
         self.duration = None
         self.result = None
         self.transaction_type = transaction_type
-        self._tracer = tracer
+        self.tracer = tracer
 
         self.dropped_spans = 0
         self.context = {}
-        self.tags = {}
 
         self.is_sampled = is_sampled
         self._span_counter = 0
+        self._span_timers = defaultdict(Timer)
+        self._duration_lock = threading.Lock()
+        try:
+            self._breakdown = self.tracer._agent._metrics.get_metricset(
+                "elasticapm.metrics.sets.breakdown.BreakdownMetricSet"
+            )
+        except (LookupError, AttributeError):
+            self._breakdown = None
+        super(Transaction, self).__init__()
 
-    def end_transaction(self):
+    def end(self, skip_frames=0):
         self.duration = _time_func() - self.start_time
+        if self._breakdown:
+            for (span_type, span_subtype), timer in compat.iteritems(self._span_timers):
+                labels = {
+                    "span.type": span_type,
+                    "span.subtype": span_subtype,
+                    "transaction.name": self.name,
+                    "transaction.type": self.transaction_type,
+                }
+                self._breakdown.timer("span.self_time", reset_on_collect=True, **labels).val = timer.val
```

Comment:
I think this should be `.update(timer.val)`? Otherwise if you have two transactions in a row, the second one will wipe out the first's timer values right?

Reply:
Great catch! Will add a test for this case.